### ISSUE TITLE
Please clarify custom AMI support documentation 

### ISSUE THREAD

**Issue by rmarable** [aws/aws-parallelcluster#1589]
The public-facing documentation of Parallelcluster's "createami" function implies that any custom AMI can be used:

- "Specifies the ID of a custom AMI to use instead of the default published AMIs."
**Source: https://docs.aws.amazon.com/parallelcluster/latest/ug/cluster-definition.html#custom-ami-section**

- "If you have a customized AMI and software already in place, you can apply the changes needed by AWS ParallelCluster on top of it."
**Source: https://docs.aws.amazon.com/parallelcluster/latest/ug/tutorials_02_ami_customization.html**

I created a new custom centos7 AMI with a root volume that is larger than 2 TB.  After generating a generic pcluster-config file, I then ran the following command to build a new pcluster-enabled AMI from my custom image:
```
$ pcluster createami --ami-id ami-*** --os centos7 -i t3.large
```
However, the resulting AMI does not appear to use the EBS volume or contain anything from my custom image (for example, there should be at least one EFS mount present):
```
$ df -h
Filesystem      Size  Used Avail Use% Mounted on
devtmpfs        463M     0  463M   0% /dev
tmpfs           495M     0  495M   0% /dev/shm
tmpfs           495M   13M  482M   3% /run
tmpfs           495M     0  495M   0% /sys/fs/cgroup
/dev/xvda1      2.0T   12G  2.0T   1% /
tmpfs            99M     0   99M   0% /run/user/1000
```

After diving deeper, it seems that we should be still be using custom Chef cookbooks to handle non-default configuration if we want to avoid the penalty of configuring the image during cluster operations which will slow down our autoscaling response.

Am I missing something obvious, or is the public documentation for `"pcluster createami"` misleading?

Thanks,

Rodney

**Comment by lukeseawalker** [comment]
Hi @rmarable,
while I'll look into this, can you please clarify

> However, the resulting AMI does not appear to use the EBS volume or contain anything from my custom image (for example, there should be at least one EFS mount present):

Do you mean that the content of the EBS root volume of the base AMI used in the createami step differ from the volume created when using the pcluster-enabled AMI?

Where/how the EFS mount was configured in the base AMI?

**Comment by lukeseawalker** [comment]
I did the following quick test:
* create an EFS
* launch an instance, mounted the EFS and configured it in the fstab
* wrote some data into the root volume
* create an image from the running instance through the console: Image -> create Image.
* use that image ID as the input of the `createami` command. The output was a new AMI id.
* use the new AMI id as the value for the parameter `custom_ami` and created a cluster.

Then I logged into the master node and I can see that the content created before in the base AMI is there and the EFS is mounted.
N.B. The EFS resource will not be managed by the ParallelCluster CloudFormation stack and it must have the right security group assigned in order to be "mountable" by the cluster itself.
